posts/aws-appsync-is-coming-for-your-api.md

```diff
@@ -0,0 +1,125 @@
+
+Once a Data Source has been added, you can then attaching custom Resolvers through the schema page. This entire setup reminds me of the old Lambda mapping behaviour, but I'm sure it'll improve over time. The resolver is a [Velocity Template](http://velocity.apache.org/engine/1.7/user-guide.html), which you use to generate a JSON request to the backend service, in this instance, DynamoDB. I can see people just ignoring them and going straight to Lambda as a proxy. To demonstrate the complexity within the template, following request-mapper is used to increment the vote counter for the `vote` mutation.
+
+```graph
+#set($allowedCandidated = ['carnival','racing','close','unicorn'] )
+#set($isAllowedCandidate= $allowedCandidated.contains($ctx.args.candidateId))
+$util.validate(${isAllowedCandidate}, "Candidated iD is not allowed")
+
+{
+    "version" : "2017-02-28",
+    "operation" : "UpdateItem",
+    "key":{
+    	"candidateId":{ "S": "${ctx.args.candidateId}" }
+    },    
+    "update" : {
+        "expression" : "ADD votes :num",
+        "expressionValues" : {
+            ":num" : { "N" : 1 }
+        }
+    }
+}
+```
+This request-mapper does a rudimentary check to ensure that only allowed `candidateIds` are submitted. If it's an appropriate candidate, it will then increment the votes field. When you get into more complex queries, or when you need to do any kind of access-control enforcement, that needs to be coded into the template, which is going to get messy. You'll want to ensure you have comprehensive API tests to ensure you don't accidentally remove your security controls.
+
+Once you're happy with your resolver (as happy as you can be), the Queries tab on the left-hand side gives you the ability to "write, validate and test GraphQL queries". It's a great way of validating your understanding of GraphQL, and also to debug your resolvers.
+
+```
+# Query Example showing how to retrieve candidates
+query preVote {
+  getCandidates {
+    candidateId,
+    votes
+  }
+}
+
+# Mutation example demonstrating how to call the mutation
+mutation iLikeUnicorns {
+  vote (candidateId: "unicorn") {
+    votes
+  }
+}
+```
+
+One of my favourite parts of the Queries page is that it actually uses the same GrapQL endpoint as your implementation. You can therefore use your browser's inspection tool to see the format of the GraphQL `POST` request if you're having trouble with your client side implementation.
+
+# Developing an Application: PonyVote
```

Comment:
You should give pointer about your application. What are the role of each component, why you designed it this way, etc. 
Probably better to do it in the README of https://github.com/ggotti/ponyvote

Reply:
I'll add it to the readme